File: Examples/python/uart_loopback.py
#!/usr/bin/python

import sys
import redpitaya_scpi as scpi

rp_s = scpi.scpi(sys.argv[1])

rp_s.tx_txt('UART:INIT')
print("Init UART")


rp_s.tx_txt('UART:BITS CS7')
print("Set bit size CS7")

rp_s.tx_txt('UART:BITS?')
print("Check bit size",rp_s.rx_txt())

rp_s.tx_txt('UART:SPEED 57600')
print("Set speed 57600")

rp_s.tx_txt('UART:SPEED?')
print("Check speed",rp_s.rx_txt())

rp_s.tx_txt('UART:STOPB STOP2')
print("Set stop bit STOP2")

rp_s.tx_txt('UART:STOPB?')
print("Check stop bit",rp_s.rx_txt())

rp_s.tx_txt('UART:PARITY ODD')
print("Set parity mode: ODD")

rp_s.tx_txt('UART:PARITY?')
print("Check parity mode",rp_s.rx_txt())

rp_s.tx_txt('UART:TIMEOUT 10')
print("Set timeout: 10 decams")

rp_s.tx_txt('UART:TIMEOUT?')
print("Check timeout",rp_s.rx_txt())


rp_s.tx_txt('UART:SETUP')
print("Setup settings")

rp_s.tx_txt('UART:WRITE7 #H11,#H22,#H33,33,33,#Q11,#B11001100')
print("Write 7 bytes to uart: #H11,#H22,#H33,33,33,#Q11,#B11001100'")

rp_s.tx_txt('UART:READ3')
print("Read: ",rp_s.rx_txt())

rp_s.tx_txt('UART:READ4')
print("Read: ",rp_s.rx_txt())

rp_s.tx_txt('UART:RELEASE')
print("Release UART")

File: Examples/Matlab/uart_loopback.m
%% Define Red Pitaya as TCP/IP object

IP= '';           % Input IP of your Red Pitaya...
port = 5000;
tcpipObj=tcpip(IP, port);

%% Open connection with your Red Pitaya

fopen(tcpipObj);
tcpipObj.Terminator = 'CR/LF';
fprintf(tcpipObj,'UART:INIT');

fprintf(tcpipObj,'UART:BITS CS7');         % set size 7 bit
res = query(tcpipObj,'UART:BITS?');        % check current settings for bit size 
fprintf('Bit size %s\n', res);

fprintf(tcpipObj,'UART:SPEED 57600');      % set uart speed
res = query(tcpipObj,'UART:SPEED?');       % check current settings for speed
fprintf('Speed %s\n', res);

fprintf(tcpipObj,'UART:STOPB STOP2');      % set stop bits
res = query(tcpipObj,'UART:STOPB?');       % check current settings for stop bits
fprintf('Stop bits %s\n', res);

fprintf(tcpipObj,'UART:PARITY ODD');       % set parity
res = query(tcpipObj,'UART:PARITY?');      % check current settings for parity
fprintf('Parity %s\n', res);

fprintf(tcpipObj,'UART:TIMEOUT 10');       % set timeout in 1/10 sec. 10 = 1 sec 
res = query(tcpipObj,'UART:TIMEOUT?');     % check current settings for parity
fprintf('Timeout %s\n', res);

fprintf(tcpipObj,'UART:SETUP');           % apply setting to uart 

fprintf(tcpipObj,'UART:WRITE7 #H11,#H22,#H33,33,33,#Q11,#B11001100');  % write to uart 7 bytes
fprintf('Write 7 bytes to uart: #H11,#H22,#H33,33,33,#Q11,#B11001100\n');

res = query(tcpipObj,'UART:READ3');        % read from uart 3 bytes
fprintf('Read: %s\n', res);

res = query(tcpipObj,'UART:READ4');        % read from uart 4 bytes
fprintf('Read: %s\n', res);

fprintf(tcpipObj,'UART:RELEASE');          % close uart

%% Close connection with Red Pitaya

fclose(tcpipObj);